python/pyarrow/includes/libarrow_fs.pxd

```diff
@@ -300,6 +300,80 @@ cdef extern from "arrow/filesystem/api.h" namespace "arrow::fs" nogil:
         const CIOContext& io_context,
         int64_t chunk_size, c_bool use_threads)
 
+cdef extern from "arrow/filesystem/hdfs_io.h" namespace "arrow::io" nogil:
```

Comment:
Are all these declarations actually needed by PyArrow?

Reply:
No, most of them aren't and are copied from `libarrow.pxd`. I can remove the unused ones - but am not sure if some external application can actually use them?